doc/docs.md

```diff
@@ -2130,6 +2103,69 @@ In this example, the `can_register` method has a receiver of type `User` named `
 The convention is not to use receiver names like `self` or `this`,
 but a short, preferably one letter long, name.
 
+### Embedded structs
+
+V supports embedded structs , similar to inherit, parent-subclasses in other programming languages.
+
+Unlike inherit in other programming languages, Embedded struct in v is just [mixin](https://en.wikipedia.org/wiki/Mixin), where there is no parent-child relationship between embedded structs, and can't type cast.
+
+```v
+struct Color {
+mut:
+        r byte
+        g byte
+        b byte
+}
+
+struct Widget {
+mut:
+        x int
+        y int
+}
+
+fn (w &Widget) area() int {
+        return w.x * w.y
+}
+
+struct Button {
+        Widget
+        Color
+        title string
+}
+```
+
+With embedding Button auto have field and method from `Color` and `Widget` do:
+
+```v oksyntax
+mut button := Button{
+	title: 'Click me'
+        y: 2
+}
+
+button.x = 3
+assert button.area() == 6
+button.r = 255
+print(button)
+```
+
+output :
+```v
+Button{
+    Widget: Widget{
+        x: 3
+        y: 2
+    }
+    Color: Color{
+        r: 255
+        g: 0
+        b: 0
+    }
+    title: 'Click me'
+}
+```
+
+Note: If multiple structures are embedded, fields and functions cannot have same name between them.
```

Comment:
It can.

You can call fields and methods of any embedded structure by specifying the structure name.

If no structure name is given, the outermost method of the same name will be called.

The same principle applies to the initialization of fields.

Basically, V embedding should works the same way as Go's embedding.

Reply:
@kahsa  Yes, you're right.

Reply:
@yuyi98 I fix the doc  see below commit , but I found you can't init a struct like this

```
mut button := Button{
  title: 'Click me'
  Widget{
    x: 4
    y: 5
  }
}
```